lib/chef/provider/package/yum.rb

```diff
@@ -1059,16 +1077,13 @@ def load_current_resource
             end
           end
 
-          # Don't overwrite an existing arch
-          unless arch
-            parse_arch
-          end
 
           @current_resource = Chef::Resource::Package.new(@new_resource.name)
           @current_resource.package_name(@new_resource.package_name)
 
           installed_version = []
           @candidate_version = []
+          @arch = []
```

Comment:
I could totally be misunderstanding this, but won't this overwrite any value set by the resource?